### PR TITLE
Fix incremental feed validation

### PR DESCRIPTION
**Pull request by talzich**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-11148

## Description
Move feed's `expirationPolicy` validation to a new  validation; `IN163`.

### CODE REVIEW

File: demisto_sdk/commands/validate/validators/IN_validators/IN163_is_valid_feed_expiration_policy.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+                elif not incremental_feed_param and not is_sudden_death(
+                    expiration_policy
+                ):
+                    invalid_content_items.append(
+                        ValidationResult(
+                            validator=self,
+                            message=self.error_message
+                            + MISSING_SUDDEN_DEATH_ERROR_MESSAGE,
+                            content_object=content_item,
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/validators/IN_validators/IN163_is_valid_feed_expiration_policy.py:110]: all of this is very long with a lot of copy paste, makes it hard to read the logic. Lets extract everything thats the same from the if statements and put at the end

so 
```
messag_in_ifs = ''
if ...
messag_in_ifs = 1
elif...
messag_in_ifs = 2....
```
...

```
contentitems.append(ValidationResult(
                            validator=self,
                            message=self.error_message
                            + **messag_in_ifs**,
                            content_object=content_item)
```

**Review by JudahSchwartz — COMMENTED** [review]
Nice! its a bit verbose lets try to cut this down a bit

File: demisto_sdk/commands/validate/validators/IN_validators/IN163_is_valid_feed_expiration_policy.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+            expiration_policy = next(
+                (
+                    param
+                    for param in content_item.params
+                    if param.name == "feedExpirationPolicy"
+                ),
+                None,
+            )
+            if content_item.is_feed:
```
[the remark this answers is not in the document]
Comment (talzich) [reply demisto_sdk/commands/validate/validators/IN_validators/IN163_is_valid_feed_expiration_policy.py:73]: done

File: demisto_sdk/commands/validate/validators/IN_validators/IN163_is_valid_feed_expiration_policy.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                if not expiration_policy:
+                    invalid_content_items.append(
+                        ValidationResult(
+                            validator=self,
+                            message=self.error_message + MISSING_EXPIRATION_POLICY,
+                            content_object=content_item,
+                        )
+                    )
+
+                elif expiration_policy.display != "" or expiration_policy.type != 17:
```
[the remark this answers is not in the document]
Comment (talzich) [reply demisto_sdk/commands/validate/validators/IN_validators/IN163_is_valid_feed_expiration_policy.py:83]: done

File: demisto_sdk/commands/validate/validators/IN_validators/IN163_is_valid_feed_expiration_policy.py (hunk 1 above)
Comment (talzich) [reply demisto_sdk/commands/validate/validators/IN_validators/IN163_is_valid_feed_expiration_policy.py]: done

File: demisto_sdk/commands/validate/validators/IN_validators/IN163_is_valid_feed_expiration_policy.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        for content_item in content_items:
+            if not content_item.is_feed:
+                continue
+
+            incremental_feed_param = next(
+                (
+                    param
+                    for param in content_item.params
+                    if param.name == "feedIncremental"
+                    and param.hidden is True
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/validators/IN_validators/IN163_is_valid_feed_expiration_policy.py:64]:
```suggestion
                    and param.hidden
```
thats redundant no?

File: demisto_sdk/commands/validate/validators/IN_validators/IN163_is_valid_feed_expiration_policy.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                    and param.type == INCREMENTAL_FEED_PARAMETER_TYPE_NUMBER
+                    and param.defaultvalue is True
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/validators/IN_validators/IN163_is_valid_feed_expiration_policy.py:66]:
```suggestion
                    and param.defaultvalue
```

**Review by JudahSchwartz — APPROVED** [review]
nice! last comment below

**Pull request merged**